app/controllers/tags_controller.rb

```diff
@@ -35,8 +35,7 @@ def edit
   # Creates a new instance of the tag.
   def create
     tag_params = params.require(:tag).permit(:name, :description)
-    new_tag = Tag.new(tag_params.merge(user: @current_user))
-
+    new_tag = Tag.new(tag_params.merge(user: @current_user, assessment: Assessment.find(params[:assignment_id])))
```

Comment:
Hmm, you don't need to do the `find` here, just do `assessment: params[:assignment_id]`.

Reply:
Hi, I tried implementing this, however, it causes some tests to fail such as: 

>   1) TagsController#create associates the new tag with a grouping when passed grouping_id
     Failure/Error: new_tag = Tag.new(tag_params.merge(user: @current_user, assessment: params[:assignment_id]))

     ActiveRecord::AssociationTypeMismatch:
       Assessment(#47369386021560) expected, got "4543" which is an instance of String(#47369333784100)
     # ./app/controllers/tags_controller.rb:40:in `create'
     # ./app/controllers/application_controller.rb:42:in `use_time_zone'
     # ./spec/controllers/tags_controller_spec.rb:37:in `block (3 levels) in <top (required)>'


  2) TagsController#create creates a new tag
     Failure/Error: new_tag = Tag.new(tag_params.merge(user: @current_user, assessment: params[:assignment_id]))

     ActiveRecord::AssociationTypeMismatch:
       Assessment(#47369386021560) expected, got "4544" which is an instance of String(#47369333784100)
     # ./app/controllers/tags_controller.rb:40:in `create'
     # ./app/controllers/application_controller.rb:42:in `use_time_zone'
     # ./spec/controllers/tags_controller_spec.rb:31:in `block (3 levels) in <top (required)>'


  3) TagsController#create associates the new tag with an assessment
     Failure/Error: new_tag = Tag.new(tag_params.merge(user: @current_user, assessment: params[:assignment_id]))

     ActiveRecord::AssociationTypeMismatch:
       Assessment(#47369386021560) expected, got "4545" which is an instance of String(#47369333784100)
     # ./app/controllers/tags_controller.rb:40:in `create'
     # ./app/controllers/application_controller.rb:42:in `use_time_zone'
     # ./spec/controllers/tags_controller_spec.rb:46:in `block (3 levels) in <top (required)>'